### ISSUE TITLE
[v2] Multiple clients fight over the traffic-manager install

### ISSUE THREAD

**Issue by rileyrohloff** [telepresenceio/telepresence#1617]
We have namespaces in our cluster for ephemeral testing environments that we'd like our developers to start using. Telepresencev2 seems like the best option, however we're now seeing that you can't have more than one connection to a cluster at a time? This completely derails the idea of having multiple developers working in a cloud environment if only one can connect at a time.

EX: ```telepresence connect --mapped-namespaces my-namespace```

Spins up a traffick-manager pod in the `ambassador` namespace.

NAME                               READY   STATUS    RESTARTS   AGE
traffic-manager-765f4bb59f-zvpd2   1/1     Running   0          83s

Then if my co-worker runs the same `connect` command for tele with a different namespace, it kills my traffic-manager session and I lose connection.

NAME                               READY   STATUS    RESTARTS   AGE
traffic-manager-765f4bb59f-zvpd2   1/1     Terminating   0          120s
traffic-manager-9doaad90ak3-zvp5   1/1     Running   0          75s

Is there not a way to have multiple connections to a cluster for different developer to work through telepresence? 

Or is there a way to install the traffic-manager into a specific namespace so that at least there can be a connection per namespace?

**Comment by rileyrohloff** [comment]
@LukeShu 

Both myself and my teammate upgraded our telepresence versions to the latest release and works perfectly! 

Thank you so much for the quick response!

**Comment by scottillogical** [comment]
I worked around this issue by installing a kyverno policy that globally locks us to a specific telepresence version (credit @MattSurabian for the idea)
```
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: require-telepresence-version
spec:
  validationFailureAction: enforce
  rules:
  - name: require-telepresence-version
    match:
      resources:
        name: traffic-manager
        namespaces:
        - ambassador
        kinds:
        - Deployment
    validate:
      message: "Only telepresence 2.2.0 is currently supported until https://github.com/telepresenceio/telepresence/issues/1617 is resolved. Install it with: sudo curl -fL https://app.getambassador.io/download/tel2/darwin/amd64/2.2.0/telepresence -o /usr/local/bin/telepresence && sudo chmod a+x /usr/local/bin/telepresence"
      pattern:
        spec:
          containers:
          - name: '*'
            Image: docker.io/datawire/tel2:2.2.0
```

**Comment by vsimon** [comment]
> The real issue here is that each of your clients is deciding that they need to upgrade the traffic manager. We should definitely improve that

@LukeShu  Could we please highlight this issue because I think coordinating the exact version among teammates is easier said than done. 

Depending on when someone runs `brew install ...` they will receive the latest version (there's currently no versioning with brew), so even if today the CI pipelines and teammates are all on the same version, when a new teammate joins the team on a later date and follows the install documentation to install via brew, they will inadvertently install a newer version and cause version churns to occur.  

This is for those that have no policy engine installed yet 😢

**Comment by josecv** [comment]
#2010 will make it so that `telepresence connect` will only result in a change to the traffic manager if the CLI is using a newer version than what is on the cluster -- I think this is a fairly reasonable solution, since it'll prevent a cycle of upgrades and downgrades. Does this sound reasonable to folks in this thread?

I suppose we ideally should also find a way to warn the user when the Traffic Manager version isn't compatible with the CLI version, but I don't think we have any way of determining that at this time.

**Comment by vsimon** [comment]
Thanks @josecv , yes, I'm supportive of it.

**Comment by efunk** [comment]
Resolved in Telepresence 2.4.1.
